scheduler/scd_utils.py

```diff
@@ -0,0 +1,239 @@
+import datetime as dt
+import collections
+import shutil
+import locale
+
+# Standardizes datetime format to use the en_US formatting in the Python docs.
+locale.setlocale(locale.LC_TIME, "en_US")
+
+class SCDUtils(object):
+    """Contains utilities for working with SCD files.
+
+    Attributes:
+        line_fmt (str): String format for scd line.
+        scd_dt_fmt (str): String format for parsing/writing datetimes.
+        scd_filename (TYPE): The filename of schedule to use.
+    """
+
+    def __init__(self, scd_filename):
+        super(SCDUtils, self).__init__()
+        self.scd_filename = scd_filename
+        self.scd_dt_fmt = "%x %H:%M"
+        self.line_fmt = "{datetime} {duration} {prio} {experiment}"
+
+    def read_scd(self):
+        """Read and parse the SCD file.
+
+        Returns:
+            TYPE: Dict
+
+        Raises:
+            ValueError: on lines with obvious errors in them.
+            Will also throw file errors if there are problems opening SCD file.
+        """
+        with open(self.scd_filename, "r") as f:
+            raw_scd = f.readlines()
+
+        raw_scd = [line.split() for line in raw_scd]
+
+        scd_lines = []
+        for num, line in enumerate(raw_scd):
+            if len(line) != 5:
+                raise ValueError("Line {} has too many arguments".format(num))
+
+            try:
+                time = dt.datetime.strptime(line[0] + " " + line[1], self.scd_dt_fmt)
+            except:
+                raise ValueError("Unable to create datetime from line {}".format(num))
+
+            duration = line[2]
```

Comment:
might want some sort of error check for duration. should be either an integer or a '-'?

Reply:
i added this.